pkg/cassandra/datacenter.go

```diff
@@ -344,6 +366,10 @@ func Coalesce(clusterName string, clusterTemplate *api.CassandraClusterTemplate,
 	// we need to declare at least one container, otherwise the PodTemplateSpec struct will be invalid
 	UpdateCassandraContainer(&dcConfig.PodTemplateSpec, func(c *corev1.Container) {})
 
+	if mergedOptions.Telemetry != nil && mergedOptions.Telemetry.Mcac != nil && mergedOptions.Telemetry.Mcac.Enabled != nil {
```

Comment:
I think we can simplify this: `dcConfig.McacEnabled` could be turned into a `bool` instead of `*bool`, then we could just:

```go
dcConfig.McacEnabled = mergedOptions.Telemetry.IsMcacEnabled()
```